src/legacy/core_plugins/visualizations/public/index.ts

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { VisTypesService, VisTypesSetup } from './vis_types';
+
+class VisualizationsPlugin {
+  private readonly visTypes: VisTypesService;
+
+  constructor() {
+    this.visTypes = new VisTypesService();
+  }
+
+  public setup() {
+    return {
+      visTypes: this.visTypes.setup(),
+    };
+  }
+
+  public stop() {
+    this.visTypes.stop();
+  }
+}
+
+/**
+ * We export visualizations here so that users importing from 'plugins/visualizations'
+ * will automatically receive the response value of the `setup` contract, mimicking
+ * the data that will eventually be injected by the new platform.
+ */
+export const visualizations = new VisualizationsPlugin().setup();
+
+/** @public */
+export interface VisualizationsSetup {
+  visTypes: VisTypesSetup;
+}
+
+/** @public types */
+export {
```

Comment:
the naming got me very confused :)

we have a VisTypesService under vis_types but here we are just re-exporting types from vis_types :) and the import on top is just importing the service and setup type

Reply:
Yeah we are exporting all of the types for all services at the top level here. So for simplicity I'm just directly exporting them. However, for the setup type, we need to compose that into the `VisualizationsSetup` interface, and it doesn't really need to be exported by itself.